connection.go

```diff
@@ -408,7 +403,7 @@ func (mc *mysqlConn) query(query string, args []driver.Value) (*textRows, error)
 	var resLen int
 	resLen, err = handleOk.readResultSetHeaderPacket()
 	if err != nil {
-		return nil, mc.markBadConn(err)
+		return nil, err
```

Comment:
This is place we shouldn't use `mc.markBadConn()` because we sent packet already.
Since `readResultSetHeaderPacket()` (and underlaying `readPacket()`) won't return `errBadConnNoWrite`, we don't have unsafe retry bug.